fpdf/fpdf.py

```diff
@@ -2438,7 +2479,25 @@ def set_link(self, link, y=0, x=0, page=-1, zoom="null"):
                 -1 indicates the current page, which is the default value.
             zoom (float): optional new zoom level after following the link.
                 Currently ignored by Sumatra PDF Reader, but observed by Adobe Acrobat reader.
+            name (str, optional): Name for the destination. If provided, creates or updates a named
+                destination in the PDF that can be referenced from other parts of the document
+                or from external documents.
         """
+        # Handle named destination case
+        if name:
+            # Create the destination
+            dest = DestinationXYZ(
+                self.page if page == -1 else page,
+                top=self.h_pt - y * self.k,
+                left=x * self.k,
+                zoom=zoom,
+            )
+            # Store it in the named destinations dictionary
+            self.named_destinations[name] = dest
+            # Return the name for reference
+            return name
+
+        # Regular link handling (backward compatibility)
```

Comment:
```suggestion
       # Handle named destination case
        if name and link is None:
            # If a name is provided but no link, get the link from the named_destinations dictionary
            # or a default dictionary if not yet set.
            link = self.named_destinations.get(name, DestinationXYZ(0, top=self.h_pt * self.k))
        else:
            link = self.links[link]
```

Then, at the end of the method:
```
      if name:
            self.named_destinations[name] = link
            return name
```